go/bucketpool/bucketpool_test.go

```diff
@@ -0,0 +1,167 @@
+package bucketpool
+
+import (
+	"math/rand"
+	"testing"
+)
+
+func TestPool(t *testing.T) {
```

Comment:
can you fuzz test this?

simple approach would be

pick random min, max, multiplier
randomly get and put, and always assert the cap(buffer) you get back is no less than what you asked for, and is no greater than what you statically compute should be the next bucket